core/trino-main/src/main/java/io/trino/sql/analyzer/StatementAnalyzer.java

```diff
@@ -3981,15 +3981,22 @@ private RelationType analyzeView(Query query, QualifiedObjectName name, Optional
                 // run view as view owner if set; otherwise, run as session user
                 Identity identity;
                 AccessControl viewAccessControl;
-                if (owner.isPresent() && !owner.get().getUser().equals(session.getIdentity().getUser())) {
+                if (owner.isEmpty()) {
+                    identity = session.getIdentity();
+                    viewAccessControl = accessControl;
+                }
+                else if (owner.get().getUser().equals(session.getIdentity().getUser())) {
                     identity = Identity.from(owner.get())
```

Comment:
why we don't need to get a list of all roles?

Reply:
The list of roles is actually provided by the access control via `io.trino.metadata.SystemSecurityMetadata#getViewRunAsIdentity` and it appears here as the `owner` `Identity`. If we get all the roles here, we kind of enforce this semantics on all access control implementations.